x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinator.java

```diff
@@ -172,6 +174,14 @@ public synchronized AutoFollowStats getStats() {
     }
 
     synchronized void updateStats(List<AutoFollowResult> results) {
+        // purge stats for removed patterns
+        var currentPatterns = this.patterns;
+        recentAutoFollowErrors.keySet().removeIf(key -> {
+            var semicolonPosition = key.indexOf(':');
+            var patternName = semicolonPosition != -1 ? key.substring(0, semicolonPosition) : key;
```

Comment:
```suggestion
            var colonPosition = key.indexOf(':');
            var patternName = colonPosition != -1 ? key.substring(0, colonPosition) : key;
```

Reply:
I think you missed this comment? Since the character we look for is a colon (':'), I think we should name the variable after that rather than the misleading semicolon name.